libindy_vdr/src/resolver/did_document.rs

```diff
@@ -173,12 +174,52 @@ fn merge_diddoc(base: &mut SJsonValue, content: &SJsonValue) {
     }
 }
 
-fn expand_verkey(id: &str, verkey: &str) -> String {
-    if let Some(stripped) = verkey.strip_prefix('~') {
-        format!("{}{}", id, stripped)
+// Returns raw verkey in case of conversion errors, otherwise 'default' indy handling
+pub fn expand_verkey(id: &str, verkey: &str) -> String {
+    let expanded_verkey = expand_verkey_indy(id, verkey);
+    if expanded_verkey.is_err() {
+        return verkey.to_string();
+    }
+    return expanded_verkey.unwrap();
+}
+
+pub fn expand_verkey_indy(id: &str, verkey: &str) -> VdrResult<String> {
+    // separate optional crypto_type from verkey part
+    let (key, key_type) = if verkey.contains(":") {
+        let vec: Vec<&str> = verkey.split(":").collect();
+        match vec.len() {
+            0 | 1 => {
+                return Err(input_err("Unexpected verkey format"));
+            }
+            _ => {
+                let key = vec[0];
+                let key_type = vec[vec.len() - 1];
+                if key.len() == 0 || key_type.len() == 0 {
+                    return Err(input_err("Unexpected verkey format"));
+                }
+                (key, Some(key_type))
```

Comment:
```suggestion
  match (vec.first(), vec.last()) {
      (key, key_type) => Ok((key, key_type)),
      _ => Err(input_err("Unexpected verkey format"))
  }
```

Might require some additional code changes as it returns a `Ok(_)` now.

Reply:
From a short test, this change would not catch empty strings and create different behavior

Reply:
Yes you are correct